File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(input: Array<string>): Number {
  const colors = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];
  let count = 0;
  for (let i = 0; i < 2; i++) {
    count += colors.indexOf(input[i]);
    if (i == 0) {
        count *= 10;
    }
  }
  return count;
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
def value(colors):
    all_colors = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
    digits = []
    for color in colors:
        digits.append(all_colors.index(color))
    number = digits[0]*10 + digits[1]
    return number